controllers/cluster_controller.go

```diff
@@ -287,3 +296,80 @@ func splitMachineList(list *clusterv1.MachineList) (*clusterv1.MachineList, *clu
 	}
 	return controlplanes, nodes
 }
+
+func (r *ClusterReconciler) reconcileControlPlaneInitialized(ctx context.Context, cluster *clusterv1.Cluster) error {
+	if cluster.Status.ControlPlaneInitialized {
+		return nil
+	}
+
+	machines, err := r.getMachinesInCluster(ctx, cluster.Namespace, cluster.Name)
+	if err != nil {
+		r.Log.Error(err, "error getting machines in cluster", "cluster", cluster.Name)
```

Comment:
We should either do `"cluster", fmt.Sprintf("%s/%s", cluster.Namespace, cluster.Name)`, or include `"namespace"` as a separate field.